mantis-server/mantis-server-agent/src/main/java/io/mantisrx/server/agent/BlobStoreAwareClassLoaderHandle.java

```diff
@@ -93,6 +93,15 @@ public UserCodeClassLoader getOrResolveClassLoader(Collection<URI> requiredFiles
         return SimpleUserCodeClassLoader.create(classLoader);
     }
 
+    @Override
+    public void cacheJobArtifacts(Collection<URI> artifacts) {
+        try {
+            getResolvedUrls(artifacts, blobStore);
+        } catch (IOException | URISyntaxException e) {
+            log.warn("Failed to download artifacts: {}. Trace: {}", artifacts, e);
```

Comment:
nit: if a Throwable is the last argument of the log line, you don't need a place holder for it.

```
log.warn("Failed to download artifacts: {}. ", artifacts, e);
```